rocm/build.sh

```diff
@@ -132,7 +139,7 @@ if (( ROCM_MAJOR_MINOR >= 507 )); then
     -GNinja \
     -DHIP_COMMON_DIR="${SCRIPT_DIR}/HIP-${ROCM_VERSION}" \
     -DCLR_BUILD_HIP=ON \
-    -DCLR_BUILD_OCL=ON \
+    -DCLR_BUILD_OCL=OFF \
```

Comment:
Is it always OK to disable this? Or should this be conditional? I don't know what this is, but this change isn't guarded behind a version test, so it would affect our existing builds, were we to rebuild them. That might be OK, I don't know: what is this?

Reply:
This option turns on building amdocl shared library, which is for running OpenCL programs. It is not used by hipclang compiler. It is turned off because ROCm 6.2/6.3 requires OpenCL ICD to be able to build it, which adds unnecessary work. Disable it should not break builds of old version of ROCm since it is optional.

Reply:
Thanks!